server/routes/users.js

```diff
@@ -0,0 +1,123 @@
+const Router = require('express')
+const User = require('../models/User')
+const router = new Router()
+const bcrypt = require('bcryptjs')
+const jwt = require('jsonwebtoken')
+
+router.post('/api/users', createUser)
+router.post('/api/user/login', logUser)
+router.get('/api/user', authenticateToken, getUserData)
+router.post('/api/user/logout', logoutUser)
+
+const createToken = (id) => jwt.sign(id.toString(), process.env.ACCESS_TOKEN)
+const maxAge = 24 * 60 * 60 * 1000
+
+async function createUser(req, res) {
+  const { password, email, name } = req.body
+
+  try {
+    const salt = await bcrypt.genSalt()
+    const hashedPassword = await bcrypt.hash(password, salt)
+
+    const users = await User.find()
+    const user = {
+      email,
+      name,
+      password: hashedPassword,
+    }
+
+    if (!users.length) {
+      user.isAdmin = true
+    }
+
+    const isExist = users.find((person) => person.email === user.email)
+
+    if (isExist) {
+      res.status(400).send({ message: 'This email is already registered' })
+    } else {
+      const newUser = new User(user)
+      await newUser.save()
+
+      const accessToken = createToken(newUser._id.toString())
+
+      res.cookie('jwt', accessToken)
+      res.status(201).json({ user: newUser._id.toString() })
+    }
+  } catch (error) {
+    const { message } = error
+
+    res.status(409).send({ message })
+  }
+}
+
+async function logUser(req, res) {
+  const { email, password } = req.body
+  const users = await User.find()
```

Comment:
@AStashevskaya Same here, you can use just `findOne` method to search user with `email`